src/modelfree/common/utils.py

```diff
@@ -149,39 +155,114 @@ def _reset_video_recorder(self, metadata=None):
         )
 
 
+class ReporterOutputFormat(KVWriter):
+    """Key-value logging plugin for Stable Baselines that writes to a Ray Tune StatusReporter."""
+    def __init__(self, reporter):
+        self.last_kvs = dict()
+        self.reporter = reporter
+
+    def writekvs(self, kvs):
+        self.last_kvs = kvs
+        self.reporter(**kvs)
+
+
 def make_session(graph=None):
     tf_config = tf.ConfigProto()
     tf_config.gpu_options.allow_growth = True
     sess = tf.Session(graph=graph, config=tf_config)
     return sess
 
 
-def simulate(venv, policies, render=False):
+class TrajectoryRecorder(object):
+    def __init__(self, num_envs, num_policies, num_traj_to_save, traj_dir):
+        self.num_envs = num_envs
+        self.num_policies = num_policies
+        self.num_traj_to_save = num_traj_to_save
+        self.traj_dir = traj_dir
+        if not os.path.isdir(self.traj_dir):
```

Comment:
Better to "to ask for forgiveness than permission" in Python for complex operations like file I/O. Although it probably doesn't matter in this particular case, in general there can be tricky race conditions if you check for file existence in operation and then do something that depends on it in a second operation. Better to just do `os.makedirs(self.traj_dir, exist_ok=True)`.

Reply:
See https://stackoverflow.com/questions/12265451/ask-forgiveness-not-permission-explain for general coding style